drivers/serial/uart_nrfx_uarte.c

```diff
@@ -0,0 +1,613 @@
+
+	nrf_gpio_pin_write(config->pseltxd, 1);
+	nrf_gpio_cfg_output(config->pseltxd);
+
+	nrf_gpio_cfg_input(config->pselrxd, NRF_GPIO_PIN_NOPULL);
+
+	nrf_uarte_txrx_pins_set(uarte,
+				config->pseltxd,
+				config->pselrxd);
+
+	if (config->hwfc == NRF_UARTE_HWFC_ENABLED) {
+		nrf_gpio_pin_write(config->pselrts, 1);
+		nrf_gpio_cfg_output(config->pselrts);
+
+		nrf_gpio_cfg_input(config->pselcts, NRF_GPIO_PIN_NOPULL);
+
+		nrf_uarte_hwfc_pins_set(uarte,
+					config->pselrts,
+					config->pselcts);
+	}
+
+	/* Configure flow control and parity checking */
+	nrf_uarte_configure(uarte,
+			    config->parity,
+			    config->hwfc);
+
+	err = baudrate_set(dev, config->baudrate);
+	if (err) {
+		return err;
+	}
+
+	/* Enable receiver and transmitter */
+	nrf_uarte_enable(uarte);
+
+	nrf_uarte_event_clear(uarte, NRF_UARTE_EVENT_ENDRX);
+
+	nrf_uarte_rx_buffer_set(uarte, &data->rx_data, 1);
+	nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STARTRX);
+
+#if CONFIG_UART_INTERRUPT_DRIVEN
+	if (interrupts_active) {
+
+		irq_enable(NRFX_IRQ_NUMBER_GET(uarte));
+
+		/* Set ENDTX event by requesting fake (zero-length) transfer.
+		 * Pointer to RAM variable (data->tx_buffer) is set because
+		 * otherwise such operation may result in HardFault or RAM
+		 * corruption.
+		 */
+		nrf_uarte_tx_buffer_set(uarte, data->tx_buffer, 0);
+		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STARTTX);
+
+		/* switch off transmitter to save an energy */
+		nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STOPTX);
+	}
+#endif
+
+	return 0;
+}
+
+#define UART_NRF_UARTE_DEVICE(idx)					      \
+	DEVICE_DECLARE(uart_nrfx_uarte##idx);				      \
+	UARTE_##idx##_CONFIG_INIT();					      \
+	static int uarte_##idx##_init(struct device *dev)		      \
+	{								      \
+		const struct uarte_init_config init_config = {		      \
+			.pseltxd = CONFIG_UART_##idx##_NRF_TX_PIN,	      \
+			.pselrxd = CONFIG_UART_##idx##_NRF_RX_PIN,	      \
+			UARTE_##idx##_NRF_HWFC_CONFIG			      \
+			.parity = UARTE_##idx##_NRF_PARITY_BIT,		      \
+			.baudrate = CONFIG_UART_##idx##_BAUD_RATE	      \
+		};							      \
+		UARTE_##idx##_INTERRUPTS_INIT();			      \
+		return uarte_instance_init(dev,				      \
+				&init_config,				      \
+				UARTE_##idx##_INTERRUPT_DRIVEN);	      \
+	}								      \
+	DEVICE_AND_API_INIT(uart_nrfx_uarte##idx,			      \
+			    CONFIG_UART_##idx##_NAME,                         \
+			    uarte_##idx##_init,                               \
+			    &uarte_##idx##_data,                              \
+			    &uarte_##idx##_config,                            \
+			    PRE_KERNEL_1,                                     \
+			    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,               \
+			    &uart_nrfx_uarte_driver_api)
+
+#define UARTE_NRF_HWFC_ENABLED(idx)			\
+	.pselcts = CONFIG_UART_##idx##_NRF_CTS_PIN,	\
+	.pselrts = CONFIG_UART_##idx##_NRF_RTS_PIN,	\
+	.hwfc = NRF_UARTE_HWFC_ENABLED,
+#define UARTE_NRF_HWFC_DISABLED				\
+	.pselcts = NRF_UARTE_PSEL_DISCONNECTED,		\
```

Comment:
Would you mind aligning these backslashes to make the code look more tidy? Sometimes they are placed with no space after the longest line, sometimes with one (and this seems to be the best option), and sometimes with way too many whitespace characters (tabs and then spaces) before them (vide `UART_NRF_UARTE_DEVICE`).

Reply:
done